File: anagramcounting.cpp
#include <bits/stdc++.h>

using namespace std;
typedef unsigned long long ll;

ll factorial(int num)
{
	ll ans = 1;
	for(int i = 1; i <= num; i++)
	{
		ans *= i;
	}
	
	return ans;
}
 
int main()
{
	string input;
	
	while(cin >> input)
	{
		map<char,int> repeated;
		for(int i = 0; i < input.size(); i++)
		{
			repeated[input[i]]++;
		}
		
		map<char,int>::iterator it;
		
		ll denominator = 1;
		for(it = repeated.begin(); it != repeated.end() ; it++)
		{
			if(it->second != 1)
				denominator *= factorial(it->second);
		}
		
		ll numerator = factorial(input.size());
		cout<<numerator/denominator<<endl;
	}
	
	return 0;
}

File: anagramcounting.py
import sys

def fact(num):
	ans = 1
	for i in range(1,num+1):
		ans = ans * i
	
	return ans
	
for _input in sys.stdin:
	s = str(_input.split()[0])
	num = int(len(s))

	lst = [0] * 60

	n = fact(num)

	for i in s:
		lst[ord(i) - ord('A')] += 1

	d = 1
	for i in lst:
		d *= fact(i)
		
	print(n//d)
	

